database/layer/admin-client.js

```diff
@@ -3,15 +3,44 @@ const pool = new PGPool()
 
 module.exports = {
   insertClient,
+  getClient,
   updateClientByClientId,
   deleteClientByClientId
 }
 
-async function insertClient (clientId, clientname, clientdescription, secret) {
-  const sql = 'INSERT INTO usher.clients (client_id, name, description, secret) VALUES ($1, $2, $3, $4)'
+/**
+ *
+ * Insert a new Client into database
+ *
+ * @param {string} tenantName Unique tenant to associate Client to
+ * @param {string} clientId The unique id for the new Client to create
+ * @param {string} clientName Client Name
+ * @param {string} clientDescription Optional description for the Client
+ * @param {string} secret
+ * @returns Record with the newly created object
+ */
+async function insertClient (tenantName, clientId, clientName, clientDescription, secret) {
   try {
-    await pool.query(sql, [clientId, clientname, clientdescription, secret])
-    return 'Insert successful'
+    // validate tenant name and get tenant key
+    let sql = 'SELECT t.key from usher.tenants t WHERE t.name = $1'
+    const tenants = await pool.query(sql, [tenantName])
+    if (tenants.rows.length === 0) {
+      throw new Error('Invalid tenant name')
```

Comment:
@mikelax This means you are requiring a client to always be associated with at least one tenant upon creation. Don't we want to keep these two entities independent of each other?

Reply:
@k1ch  yes, my thinking is that there really isn't a use case where we would want a client to exist that isn't attached to at least one tenant, so might as well force this at time of create. I think this type of functionality is fine as this is the typical use case. Otherwise we are requiring the caller to make two APIs calls in succession. 

The similar functional question will be when we implement an API to remove a client from a tenant (not necessarily delete a Client) do we enforce a Client to be attached to at least one Tenant.